File: Maximum Length[ 11-April-2023]/solution.cpp
//  >> C++ CODE <<

int max_length(int a, int b, int c) {
    int max_val = max(a, max(b, c));
    int y;
    if (max_val == a) {
        y = b + c;
    } else if (max_val == b) {
        y = c + a;
    } else {
        y = a + b;
    }
    if (2 * (y + 1) >= max_val) {
        return a + b + c;
    }
    return -1;
}

File: Maximum Length[ 11-April-2023]/solution.java
// >> JAVA CODE <<

int max=Math.max(a,Math.max(b,c));
        
        int y;
        
        if(max==a) y=b+c;
        else if(max==b) y=c+a;
        else y=a+b;
        
        
        if(2*(y+1)>=max) return a+b+c;
        
            return -1;